src/std/sugar.ss

```diff
@@ -300,3 +301,49 @@
 
   ((_ (v . more) (acc ...) . previous) ; previous is not set after <> was replaced
    (~chain-aux-diamond more (acc ... v) . previous)))
+
+;; is converts a given value into a predicate testing for the presence of the
+;; given value. Optionally a transforming procedure can prefix the value, which
+;; can in this case also be a procedure. This allows to 'get' a value out of a
+;; compound data structure before comparison (first map, then test).
+;; For numbers, char and string specialized procedures are used automatically
+;; if passed to the macro as value and not as variable. Alternatively, the
+;; test: keyword can be used to supply a test, the default is equal?.
+;;
+;; Example:
+;;  (find (is cdr 5) '((a . 2) (b . 5) (c . 6)))
+;; => (b . 5)
+;;
+;;  (filter (is file-type 'regular) (directory-files))
+;; => ("Documents" "Pictures" "Videos" "Music")
+(defrules is ()
+  ((_ proc n)
+   (stx-number? #'n)
+   (lambda (v) (= n (proc v))))
```

Comment:
Actually we have the same problem here as well, if the supplied value is not a number it will fail with an exception.

Reply:
So we need to test if the supplied value is a number before applying `=`.